gtda/plotting/persistence_diagrams.py

```diff
@@ -26,61 +26,71 @@ def plot_diagram(diagram, homology_dimensions=None, **input_layout):
     if homology_dimensions is None:
         homology_dimensions = np.unique(diagram[:, 2])
 
-    max_filt_param = np.where(np.isinf(diagram), -np.inf, diagram).max()
+    diagram_no_dims = diagram[:, :2]
+    max_birth, max_death = np.where(
+        np.isposinf(diagram_no_dims), -np.inf, diagram_no_dims
+    ).max(axis=0)
+    min_birth, min_death = np.where(
+        np.isneginf(diagram_no_dims), np.inf, diagram_no_dims
+    ).min(axis=0)
 
-    layout = dict(
+    fig = gobj.Figure()
+    fig.add_trace(gobj.Scatter(x=[100 * min(-np.abs(max_death), min_birth),
+                                  100 * max_death],
+                               y=[100 * min(-np.abs(max_death), min_birth),
+                                  100 * max_death],
+                               mode='lines',
+                               line=dict(dash='dash', width=1, color='black'),
+                               showlegend=False, hoverinfo='none'))
+
+    for dim in homology_dimensions:
+        name = f'H{int(dim)}' if dim != np.inf else 'Any homology dimension'
+        subdiagram = _subdiagrams(np.asarray([diagram]), [dim],
+                                  remove_dim=True)[0]
+        diff = (subdiagram[:, 1] != subdiagram[:, 0])
+        subdiagram = subdiagram[diff]
+        fig.add_trace(gobj.Scatter(x=subdiagram[:, 0], y=subdiagram[:, 1],
+                                   mode='markers', name=name))
+
+    range = max_death - min_birth
+    extra_space = 0.02 * range
+
+    fig.update_layout(
         width=500,
         height=500,
         xaxis1=dict(
             title='Birth',
             side='bottom',
             type='linear',
-            range=[0, 1.1 * max_filt_param],
+            range=[min_birth - extra_space, max_death + extra_space],
+            autorange=False,
             ticks='outside',
-            anchor='y1',
             showline=True,
             zeroline=True,
+            linewidth=1,
+            linecolor='black',
+            mirror=False,
             showexponent='all',
             exponentformat='e'
         ),
         yaxis1=dict(
             title='Death',
             side='left',
             type='linear',
-            range=[0, 1.1 * max_filt_param],
+            range=[min_birth - extra_space, max_death + extra_space],
```

Comment:
Why fix the range of both axes to the same? I was thinking we could use the range for each of the axes separately.

Reply:
Because then the diagonal would be oddly placed (e.g. the plot would look like a tall rectangle with a 45 degree line on the bottom right-hand corner.  I think people sort of expect their persistence diagrams to be square...  But maybe I'm missing your point?

Reply:
No, it's spot on!
My comment was about the possibility of diverging from the common expectations. But it's ok if it's a design choice !